kart/schema.py

```diff
@@ -237,18 +239,25 @@ def interpretation(self):
         return self.get("interpretation")
 
     @property
-    def id_or_name_or_interpretation(self):
+    def best_identifier(self):
         """
         Different types of schemas have these various ways of identifying themselves.
         When we do schema diffs, it makes them more readable if we match up the columns using this property.
+        At a last resort, we use the hash of the entire column contents as an "ID".
         """
         if "id" in self:
             return self.get("id")
         if "name" in self:
             return self.get("name")
         if "interpretation" in self:
             return self.get("interpretation")
-        return None
+        return self._id_from_hash
```

Comment:
As you say below, interpretation information might be missing. In some cases this may be by design - if the raw data has, for whatever reason, a dozen bands, there's no way to assign them all a different color interpretation, so presumably most or all won't be assigned one - it will be up to user to know how to interpret each band

The schema-extraction code is probably a bit bare bones - there can be more details we are missing eg band description, and of course color tables / category tables. I'll put a note in the bug